src/integer/mat_z/arithmetic/div.rs

```diff
@@ -0,0 +1,292 @@
+// Copyright © 2023 Marcel Luca Schmidt and Sven Moog
+//
+// This file is part of qFALL-math.
+//
+// qFALL-math is free software: you can redistribute it and/or modify it under
+// the terms of the Mozilla Public License Version 2.0 as published by the
+// Mozilla Foundation. See <https://mozilla.org/en-US/MPL/2.0/>.
+
+//! Implementation of scalar multiplication for [`MatZ`] matrices.
+
+use super::super::MatZ;
+use crate::integer::Z;
+use crate::macros::arithmetics::{
+    arithmetic_trait_borrowed_to_owned, arithmetic_trait_mixed_borrowed_owned,
+};
+use crate::rational::MatQ;
+use crate::traits::{GetNumColumns, GetNumRows};
+use flint_sys::fmpq_mat::fmpq_mat_set_fmpz_mat_div_fmpz;
+use flint_sys::fmpz_mat::fmpz_mat_scalar_divexact_fmpz;
+use std::ops::Div;
+
+impl Div<&Z> for &MatZ {
+    type Output = MatQ;
+    /// Implements division for a [`MatZ`] matrix by a [`Z`] integer.
+    ///
+    /// Parameters:
+    /// - `divisor`: specifies the divisor by which the matrix is divided
+    ///
+    /// Returns the product of `self` and `other` as a [`MatQ`] or panics if
+    /// the divisor is 0.
+    ///
+    /// # Examples
+    /// ```
+    /// use qfall_math::integer::{MatZ, Z};
+    /// use std::str::FromStr;
+    ///
+    /// let mat1 = MatZ::from_str("[[3,6],[9,27]]").unwrap();
+    /// let integer = Z::from(3);
+    ///
+    /// let mat2 = &mat1 / &integer;
+    /// ```
+    ///
+    /// # Panics ...
+    /// - if the divisor is `0`.
+    fn div(self, divisor: &Z) -> Self::Output {
+        if divisor == &Z::ZERO {
+            panic!("DivisionByZero: tried to divide {} by zero", self);
+        }
+
+        let mut out = MatQ::new(self.get_num_rows(), self.get_num_columns());
+        unsafe {
+            fmpq_mat_set_fmpz_mat_div_fmpz(&mut out.matrix, &self.matrix, &divisor.value);
+        }
+        out
+    }
+}
+
+arithmetic_trait_borrowed_to_owned!(Div, div, MatZ, Z, MatQ);
+arithmetic_trait_mixed_borrowed_owned!(Div, div, MatZ, Z, MatQ);
+
+impl MatZ {
+    /// Implements division for a [`MatZ`] matrix by a [`Z`] integer.
+    ///
+    /// Parameters:
+    /// - `divisor`: specifies the divisor by which the matrix is divided
+    ///
+    /// Returns the product of `self` and `other` as a [`MatZ`] or panics if
+    /// the divisor is 0.
+    ///
+    /// # Safety
+    /// The divisor MUST exactly divide each element in the matrix.
+    /// If this is not the case, the result can contain arbitrary values
+    /// which can depend on the location in memory.
+    ///
+    /// # Examples
+    /// ```
+    /// use qfall_math::integer::{MatZ, Z};
+    /// use std::str::FromStr;
+    ///
+    /// let mut mat1 = MatZ::from_str("[[3,6],[9,27]]").unwrap();
+    /// let integer = Z::from(3);
+    ///
+    /// let mat2 = unsafe{ mat1.div_exact(&integer)};
+    /// ```
```

Comment:
1. You can use 3 directly, you don't need to create a Z first.
2. Please add an assert so that one can see how the result looks like.

Reply:
> Please add an assert so that one can see how the result looks like.

This also applies to the other doc tests.